packages/perseus/src/util.ts

```diff
@@ -919,6 +919,23 @@ const unescapeMathMode: (label: string) => string = (label) =>
 
 const random: RNG = seededRNG(new Date().getTime() & 0xffffffff);
 
+// TODO(benchristel): in the future, we may want to make deepClone work for
+// Record<string, Cloneable> as well. Currently, it only does arrays.
+type Cloneable =
+    | null
+    | undefined
+    | boolean
+    | string
+    | number
+    | Cloneable[]
+    | readonly Cloneable[];
+function deepClone<T extends Cloneable>(obj: T): T {
```

Comment:
NAN: This makes me think we need to switch from Underscore to Lodash - Lodash has a `cloneDeep` helper, though I'm not sure how well it works with TS.

Reply:
Eventually we will be able to use the browser-native `structuredClone`.